extensions/parquet/base/src/main/java/io/deephaven/parquet/base/RowGroupReaderImpl.java

```diff
@@ -34,14 +37,21 @@ final class RowGroupReaderImpl implements RowGroupReader {
             @NotNull final RowGroup rowGroup,
             @NotNull final SeekableChannelsProvider channelsProvider,
             @NotNull final URI rootURI,
-            @NotNull final MessageType type,
             @NotNull final MessageType schema,
             @Nullable final String version) {
         this.channelsProvider = channelsProvider;
         this.rowGroup = rowGroup;
         this.rootURI = rootURI;
-        this.type = type;
-        for (ColumnChunk column : rowGroup.columns) {
+        this.schema = schema;
+        schemaMap = new HashMap<>(schema.getFieldCount());
```

Comment:
You can store schema.getFieldCount() in a variable and use that in following lines as well as for a size check in place of the check before line 52 instead of line 72 to fail early.